src/js/components/Dashboard.jsx

```diff
@@ -0,0 +1,94 @@
+import {connect} from 'react-redux'
+import {fetchVisualizationResults} from '../modules/visualization-results'
+import FilterCriteria from './FilterCriteria'
+import {GridList} from 'material-ui'
+import {verticalTop} from '../styles/common'
+import Visualization from './Visualization'
+import React, {Component, PropTypes} from 'react'
+
+const styles = {
+  gridList: {
+    width: '100%',
+    height: '60%',
+    overflowY: 'auto',
+    marginBottom: 24
+  },
+  visualization: {
+    height: '100%'
+  }
+}
+
+class Dashboard extends Component {
+  static propTypes = {
+    dashboard: PropTypes.object.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    filters: PropTypes.array.isRequired
+  }
+  
+  onClickFilter () {
+    const {dispatch, filters} = this.props
+    const {gridList} = this.refs
+    
+    for (const visualizationComponent of gridList.props.children) {
+      const {visualization = {}} = visualizationComponent.props
+      
+      dispatch(fetchVisualizationResults(visualization._id, filters))
```

Comment:
we need to check to see if the `_id` property exists first. If it doesn't we will likely be trying to do a fetch on undefined. also, i would use the `React.Children` utilities to iterate children

```javascript
onClickFilter () {
  const {children, dispatch, filters} = this.props
  const {gridList} = this.refs

  React.Children.forEach(children, (child) => {
    const {_id = null} = child.props.visualization

    if (!_id) {
      continue
    }

    dispatch(fetchVisualizationResults(_id, filters))
  })
```